src/hats/catalog/dataset/table_properties.py

```diff
@@ -0,0 +1,163 @@
+    """Column name in the joining (right) side of join"""
+
+    join_column_association: Optional[str] = Field(default=None, alias="hats_col_assn_join_assn")
+    """Column name in the association table that matches the joining (right) side of join"""
+
+    contains_leaf_files: Optional[bool] = Field(default=None, alias="hats_assn_leaf_files")
+    """Whether or not the association catalog contains leaf parquet files"""
+
+    indexing_column: Optional[str] = Field(default=None, alias="hats_index_column")
+    """Column that we provide an index over"""
+
+    extra_columns: Optional[List[str]] = Field(default=None, alias="hats_index_extra_column")
+    """Any additional payload columns included in index"""
+
+    ## Allow any extra keyword args to be stored on the properties object.
+    model_config = ConfigDict(extra="allow", populate_by_name=True, use_enum_values=True)
+
+    @field_validator("extra_columns", mode="before")
+    @classmethod
+    def space_delimited_list(cls, str_value: str) -> List[str]:
+        """Convert a space-delimited list string into a python list of strings."""
+        if isinstance(str_value, str):
+            if str_value:
+                str_value = str_value.strip()
+            if str_value:
+                return str_value.split(" ")
+        return str_value
+
+    @field_serializer("extra_columns")
+    def serialize_space_delimited_list(self, str_list) -> str:
+        """Convert a python list of strings into a space-delimited string."""
+        return " ".join(str_list)
+
+    @model_validator(mode="after")
+    def check_allowed_and_required(self) -> Self:
+        """Check that type-specific fields are appropriate, and required fields are set."""
+        explicit_keys = set(
+            self.model_dump(by_alias=False, exclude_none=True).keys() - self.__pydantic_extra__.keys()
+        )
+
+        allowed_keys = set(
+            CATALOG_TYPE_ALLOWED_FIELDS[self.catalog_type] + ["catalog_name", "catalog_type", "total_rows"]
+        )
+        non_allowed = explicit_keys - allowed_keys
+        if len(non_allowed) > 0:
+            raise ValueError(f"Unexpected property for table type {self.catalog_type} ({non_allowed})")
+
+        required_keys = set(
+            CATALOG_TYPE_REQUIRED_FIELDS[self.catalog_type] + ["catalog_name", "catalog_type", "total_rows"]
+        )
+        missing_required = required_keys - explicit_keys
+        if len(missing_required) > 0:
+            raise ValueError(
+                f"Missing required property for table type {self.catalog_type} ({missing_required})"
+            )
+        return self
+
+    def explicit_dict(self):
+        """Create a dict, based on fields that have been explicitly set, and are not "extra" keys."""
+        explicit = self.model_dump(by_alias=False, exclude_none=True, round_trip=True)
+        extra_keys = self.__pydantic_extra__.keys()
+        explicit = {(key, val) for key, val in explicit.items() if key not in extra_keys}
```

Comment:
```suggestion
        explicit = {key: val for key, val in explicit.items() if key not in extra_keys}
```
and then the conversion to `dict()` on the following line isn't needed.

Reply:
Thanks!